### ISSUE TITLE
[Document] GrapeJS Editable

### ISSUE THREAD

**Issue by brusch** [pimcore/pimcore#7729]
https://grapesjs.com/

**Comment by ChrisB9** [comment]
For this, I already implemented a prototype https://github.com/ChrisB9/PimcorePageBuilderBundle
I currently can't really continue building this, because of limited time and lacking customer support.
One thing, that I stumbled upon, making this so tremendously difficult, isn't the integration part but making it viable for a regular user to use.
Questions I ran into after building it for a customer:
- What theme do you use
- How will it look like in the frontend for non-enterprise businesses
- How will it be customizable for enterprises
- How do you hide the necessity to know what "margin", "padding", "classes" etc are.

In the end, grapesjs is an amazing tool, but definitely has its shortcomings that need to be picked.

**Comment by brusch** [comment]
@ChrisB9 Thanks for your quick feedback. Yes, I'm aware of your bundle, cool stuff already! I know that there are some challenges to takle, I've created this issue mainly as a start & discussion point.

**Comment by ChrisB9** [comment]
:+1: 
Yes, I'd love to provide further ideas and problems I had while building this, If you plan on going forward with this.
If therefore have any questions, feel free to ask

**Comment by brusch** [comment]
@ChrisB9 Great, thanks a lot!

**Comment by NiklasBr** [comment]
I would be much more in favour of [Gutenberg](https://github.com/WordPress/gutenberg/) it feels much more mature in use. But GrapeJs would be a significant improvement anyway.

**Comment by ChrisB9** [comment]
Yes, Gutenberg would be great, because it is supported by Drupal and WordPress and removes the limitation of lacking the understanding by normal clients. Plus, this is pretty neat: https://gutenbergcloud.org/blocks
It comes with the burden though of putting another heavy performance trade off into the editor. It now has to load ReactJS and ExtJS. And if there are multiple editors, then its multiple ReactJS-instances..

**Comment by alexej-d** [comment]
+1 for Gutenberg: It comes with a maturing community and lots of complete implementations Pimcore could simply reuse – no need to reinvent the wheel. Also this would make the jump to Pimcore for WordPress folks much easier which would lead to a much better adaption rate 👌

**Comment by dpfaffenbauer** [comment]
@alexej-d I agree that gutenberg might be a good product. but I don't agree with the rest you said... wordpress folks would not just adopt cause of gutenberg... it would't improve adaption rate at all.... it would just change Pimcore.

**Comment by alexej-d** [comment]
@dpfaffenbauer well I can probably only speak for myself as I come from a predominantly WordPress-laden background and left some years ago. But because of Gutenberg I am tempted to start building more products with WP again. On the other hand I would not even thought about touching WordPress again, if Pimcore had a solution like Gutenberg with even half as much prebuilt Blocks from a mature community.

Maybe you can elaborate as to why you think WordPress people would not adopt if they needed something better than WP?

**Comment by dpfaffenbauer** [comment]
I am saying that Gutenberg for sure is no reason to migrate to something else. It is the whole ecosystem, the community, the workflow, the company you work in or for. It has a thousands of reasons to adopt to another system, and it is for sure not a WYSIWYG Editor like Gutenberg. BTW: Comparing Wordpress to Pimcore is comparing a Starship to a Ferris Wheel, per se not comparable.

**Comment by ChrisB9** [comment]
i do agree with @dpfaffenbauer, in my experience Gutenberg has been a pain in some of my clients websites too and i suspect they would migrate just because of that. So far, my clients also never cared too much what kind of editor is in place. They want a set of blocks they can use and configure without much technical knowledge. I even realized, that pimcores areablocks are already doing a really good job. IMO lacking only nesting, some kind of grid-system drag and drop and a library to install lots of 3rd party ones

**Comment by NiklasBr** [comment]
Any news on this initiative?

**Comment by solverat** [comment]
@NiklasBr I guess this obsolete due to the upcoming new backend ui i guess...

**Comment by NiklasBr** [comment]
If the new backend UI will have some way to visually edit bricks and content in a WYSIWYG-like fashion it would need to integrage some kind of library like GrapeJS or Gutenberg, no? Or will the new UI just re-use the current old way of doing thing? I suppose something developed in-house could be done...
